bokehjs/src/coffee/models/sources/column_data_source.coffee

```diff
@@ -5,21 +5,27 @@ import * as hittest from "../../core/hittest"
 import {SelectionManager} from "../../core/selection_manager"
 import {logger} from "../../core/logging"
 import * as p from "../../core/properties"
+import {decode_column_data} from "../../core/util/deserialize"
 
 # Datasource where the data is defined column-wise, i.e. each key in the
 # the data attribute is a column name, and its value is an array of scalars.
 # Each column should be the same length.
 export class ColumnDataSource extends DataSource
   type: 'ColumnDataSource'
 
+  initialize: (options) ->
+    super(options)
+    [@data, @_shapes] = decode_column_data(@data)
+
   @define {
-      data:              [ p.Any,      {} ]
-      column_names:      [ p.Array,    [] ]
+      data:         [ p.Any,   {} ]
+      column_names: [ p.Array, [] ]
     }
 
   @internal {
     selection_manager: [ p.Instance, (self) -> new SelectionManager({source: self}) ]
     inspected:         [ p.Any ]
+    _shapes:           [ p.Any, {}]
```

Comment:
I would like to have a comment with exact type of `_shape`. It's a shame that `inspected` doesn't show the type and encourage to follow the pattern.